packages/entities/entities-gateway-services/src/composables/useUrlValidators.ts

```diff
@@ -0,0 +1,84 @@
+import type { ProtocolItem } from 'src/types'
+import composables from '.'
+
+export default function useUrlValidators() {
+  const { i18n: { t } } = composables.useI18n()
+
+  const validateHost = (host: string): string => {
+  // check if host is empty
+    if (!host || host.trim() === '') return t('gateway_services.form.errors.host.empty')
+
+    // check if a valid host (domain or ip address)
+    const hostnameRegex = /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])$|^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/
```

Comment:
q: did you confirm if this regex is reliable?

Reply:
yes, i tested it pretty well. il write some unit test case